sdk/schemaregistry/azure-schemaregistry-client/src/main/java/com/azure/schemaregistry/client/CachedSchemaRegistryClientBuilder.java

```diff
@@ -0,0 +1,284 @@
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * Builder implementation for {@link CachedSchemaRegistryClient}.
+ */
+@ServiceClientBuilder(serviceClients = CachedSchemaRegistryClient.class)
+public class CachedSchemaRegistryClientBuilder {
+    private final ClientLogger logger = new ClientLogger(CachedSchemaRegistryClientBuilder.class);
+
+    private static final String DEFAULT_SCOPE = "https://eventhubs.azure.com/.default";
+    private static final String CLIENT_PROPERTIES = "azure-schemaregistry-client.properties";
+    private static final String NAME = "name";
+    private static final String VERSION = "version";
+
+    private final String schemaRegistryUrl;
+    private final HashMap<String, Function<String, Object>> typeParserDictionary;
+    private final List<HttpPipelinePolicy> policies;
+    private final String clientName;
+    private final String clientVersion;
+    private final HttpHeaders headers;
+
+    private HttpClient httpClient;
+    private int maxSchemaMapSize;
+    private TokenCredential credential;
+    private HttpLogOptions httpLogOptions;
+    private HttpPipeline httpPipeline;
+    private RetryPolicy retryPolicy;
+
+    /**
+     * Sets the service endpoint for the Azure Schema Registry instance.
+     * Supplies client defaults.
+     *
+     * @param schemaRegistryUrl The URL of the Azure Schema Registry instance
+     * @throws NullPointerException if {@code schemaRegistryUrl} is null
+     * @throws IllegalArgumentException if {@code schemaRegistryUrl} cannot be parsed into a valid URL.
+     */
+    public CachedSchemaRegistryClientBuilder(String schemaRegistryUrl) {
+        Objects.requireNonNull(schemaRegistryUrl, "'schemaRegistryUrl' cannot be null.");
+
+        try {
+            new URL(schemaRegistryUrl);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsWarning(
+                new IllegalArgumentException("'schemaRegistryUrl' must be a valid URL.", ex));
+        }
+
+        if (schemaRegistryUrl.endsWith("/")) {
+            this.schemaRegistryUrl = schemaRegistryUrl.substring(0, schemaRegistryUrl.length() - 1);
+        } else {
+            this.schemaRegistryUrl = schemaRegistryUrl;
+        }
+
+        this.policies = new ArrayList<>();
+        this.httpLogOptions = new HttpLogOptions();
+        this.maxSchemaMapSize = CachedSchemaRegistryClient.MAX_SCHEMA_MAP_SIZE_DEFAULT;
+        this.typeParserDictionary = new HashMap<>();
+        this.httpClient = null;
+        this.credential = null;
+        this.retryPolicy = new RetryPolicy("retry-after-ms", ChronoUnit.MILLIS);
+
+        Map<String, String> properties = CoreUtils.getProperties(CLIENT_PROPERTIES);
+        clientName = properties.getOrDefault(NAME, "UnknownName");
+        clientVersion = properties.getOrDefault(VERSION, "UnknownVersion");
+
+        this.headers = new HttpHeaders();
```

Comment:
Is this ever mutated? If there are plans to add static headers let's leave it, but otherwise we can just remove it.

Reply:
Ok let's take it out for now.  We had some ideas on adding headers for internal testing, but we don't have any real use cases for clients adding their own headers to HTTP calls.